src/main/java/org/jboss/xavier/integrations/jpa/service/AnalysisService.java

```diff
@@ -117,4 +119,13 @@ public void updateStatus(String status, Long id) {
         analysisModel.setLastUpdate(new Date());
         analysisRepository.save(analysisModel);
     }
+
+    public void markAsFailedIfNotCreated(Long id) {
+        AnalysisModel analysisModel = findById(id);
+        if (!STATUS.CREATED.toString().equalsIgnoreCase(analysisModel.getStatus())) {
```

Comment:
Maybe it would better to apply the selection criteria directly in the query with something like `findByIdAndStatusIsNot(Long id, String status)` so that we can leverage the SQL condition.

Reply:
+1